src/components/Microsoft.DotNet.UpgradeAssistant/Checks/BypassableReadinessCheck.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Checks
+{
+    public abstract class BypassableReadinessCheck : IUpgradeReadyCheck
```

Comment:
Having it an abstract class requires callers to pass stuff through. I'd prefer an additional interface that can be mapped to this check (ie IBypassableReadinessCheck)

Reply:
Removed